code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1332,3 +1332,11 @@ var/list/cyborg_list = list()
 /mob/living/silicon/robot/proc/toggle_modulelock()
 	modulelock = !modulelock
 	return modulelock
+
+//Currently only used for borg movement, to avoid awkward situations where borgs with RTG or basic cells are always slowed down
+/mob/living/silicon/robot/proc/get_percentage_power_for_movement()
+	if(get_cell())
+		var/percentage = round(cell.maxcharge / 4)
```

Comment:
calling `get_cell` to check if it exists and then accessing `cell` directly later? what's the point

Reply:
lemme fix